### ISSUE TITLE
[httpbis: 0.9.0] Compilation error on windows 10

### ISSUE THREAD

**Issue by dkijania** [stepancheg/rust-http2#41]
compilation error:

```
error[E0053]: method `connect` has an incompatible type for trait
   --> C:\Users\Dariusz\.cargo\registry\src\github.com-1ecc6299db9ec823\httpbis-0.9.0\src\socket_unix.rs:158:10
    |
158 |     ) -> Pin<Box<dyn Future<Output = io::Result<Box<dyn StreamItem + Send>>> + Send>> {
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `std::pin::Pin`, found struct `std::boxed::Box`
    |
   ::: C:\Users\Dariusz\.cargo\registry\src\github.com-1ecc6299db9ec823\httpbis-0.9.0\src\socket.rs:103:10
    |
103 |     ) -> Pin<Box<dyn Future<Output = io::Result<Pin<Box<dyn StreamItem + Send>>>> + Send>>;
    |          --------------------------------------------------------------------------------- type in trait
    |
    = note: expected fn pointer `fn(&socket_unix::SocketAddrUnix, &tokio::runtime::handle::Handle) -> std::pin::Pin<std::boxed::Box<(dyn core::future::future::Future<Output = std::result::Result<std::pin::Pin<std::boxed::Box<(dyn socket::StreamItem + std::marker::Send + 'static)>>, std::io::Error>> + std::marker::Send + 'static)>>`
               found fn pointer `fn(&socket_unix::SocketAddrUnix, &tokio::runtime::handle::Handle) -> std::pin::Pin<std::boxed::Box<(dyn core::future::future::Future<Output = std::result::Result<std::boxed::Box<(dyn socket::StreamItem + std::marker::Send + 'static)>, std::io::Error>> + std::marker::Send + 'static)>>`
error: aborting due to previous error
For more information about this error, try `rustc --explain E0053`.
error: could not compile `httpbis`.
```

Fix:

`src/socket_unix.rs`:
```
158: -    ) -> Pin<Box<dyn Future<Output = io::Result<Box<dyn StreamItem + Send>>> + Send>> {
159: +    ) -> Pin<Box<dyn Future<Output = io::Result<Pin<Box<dyn StreamItem + Send>>>> + Send>> {
```

**Comment by stepancheg** [comment]
Thanks!

**Comment by stepancheg** [comment]
Published 0.9.1.

**Comment by stepancheg** [comment]
Added build on Windows in CI: https://github.com/stepancheg/rust-http2/pull/43/files
